website/docs/developer/research/research-lambda-container.md

```diff
@@ -148,6 +148,156 @@ Based on this research, it looks like using lambda as containers is possible whe
 
 If the above issues are solved, then implementing lambda as container is recommended. One way to use private ECR repo is to use a script or terraform code if possible that will create the ECR repo and then build, tag and push the container image to that repo. One that is done, update the terraform lambda modules and deploy the lambda. An additional [card](https://bugs.earthdata.nasa.gov/browse/ORCA-375) has been created to look into this way.
 
+
+### Modifying get_current_archive_list and perform_orca_reconcile
+
+The Orca Internal Reconciliation workflow lambdas require an alternative approach.
+- The maximum time limit for lambdas is 15 minutes. These lambdas may take a significant amount of time, and should not be subject to this limitation.
+
+- Code changes
+  - Merge `get_current_archive_list` and `perform_orca_reconcile` into one codebase.
+  - Wrap functionality in a loop that will process the internal-report queue until no entries remain.
+  - Since ECS does not support timeout, create an overarching timing mechanic that exits if an infinite loop occurs while processing a queue entry.
+    - Alternatively, a side-program could manually stop the task if it exceeds its' time limit.
+    - Remember that in addition to processing time, Aurora Serverless can take up to 5 minutes to spin up.
+  - Raise the internal-report queue's `visibility_timeout_seconds` to the expected timeout.
+  - Environment variables can be passed in at task definition, or when the task is run. The former should be sufficient.
+
+- Use an alternative Dockerfile. The example below packages two code files into a lightweight python container with a `CMD` to run the main file.
+  ```yaml
+  FROM python:3.8-slim-buster
```

Comment:
There are some best practices and some hardening you could put into here for your Dockerfile. Consider the following example code below where we leverage an intermediate build image, create and run as a non root user, and put items that do not change frequently at the top of the build, and those that change more frequently at the bottom. Changes like those proposed lead to a better security (running as non-root so fewer permissions needed) and smaller images (fewer layers and smaller footprint since dev libraries removed). 

```
# Build Stage - Here we can bring dev libraries and compile the wheels for the python libs
# =============================================================================
FROM python:3.8-slim as builder

WORKDIR /app

ENV PYTHONDONTWRITEBYTECODE 1
ENV PYTHONUNBUFFERED 1

# Install compiler and libraries needed for database drivers
RUN apt-get update && \
    apt-get install -y --no-install-recommends gcc libpq-dev libc6-dev

# Install python app requirements
COPY requirements.txt .
RUN pip wheel --no-cache-dir --no-deps --wheel-dir /app/wheels -r requirements.txt


# Final Stage -- This will be the final image
# =============================================================================
FROM python:3.8-slim

# Set metadata information for the image
LABEL application.name="ORCA Internal Reconciliationl" \
    application.description="Reconciles ORCA holdings with what is actually in the S3 Glacier Archive" \
    application.team="ORCA" \
    application.version="v0.0.1"

# CREATE Non-Privileged User
RUN mkdir -p /app \
    && groupadd -r appuser \
    && useradd -r -s /bin/false -g appuser appuser \
    && chown -R appuser:appuser /app

# Install libpq C library for postgres
RUN apt-get update && \
    apt-get install -y --no-install-recommends libpq5

# Copy compiled wheels and install the requirements
COPY --from=builder /app/wheels /wheels
RUN pip install --no-cache /wheels/*

# Set the work directory for our application
WORKDIR /app

# Copy our application source files
COPY *.py .

# Set the User that the app will run as
USER appuser

# Set the entry point for the container
ENTRYPOINT ["python", "main.py"]
```

Reply:
Added